### DISCUSSION TITLE
Takaajattomat taattavat

### DISCUSSION THREAD

**Discussion by emta001** [KohaSuomi/Koha#802] [Tarkistuspyynnöt]
OUTIssa törmättiin jokin aika sitten tilanteeseen, jossa takaajaton lapsiasiakas esti muiden lapsiasiakaiden categorycoden päivittymisen LAPSI-asiakkaasta LAOMATOIMI-asiakkaaksi. Tämän tutkimisen yhteydessä selvisi, että OUTIssa on useampikin asiakas, joilla tulisi olla takaaja muttei ole. Näitä tapauksia löytyy myös muista kimpoista (paitsi Siilistä). Syynä on oletetettavasti bugi ajalta ennen vuoden 2022 versionvaihtoa.

OUTIn ongelma ei onneksi koske muita kimppoja, joissa LAPSI-asiakkaat päivitetään suoraan HENKILÖ-asiakkaiksi. Mutta takaajattomuus voi aiheuttaa muunlaisia ongelmia, esimerkiksi lainatessa, joten nämä olisi hyvä käydä läpi.

Seuraavalla raportilla saa selville takaajattomat taattavat:
```
SELECT borrowernumber, categorycode 
FROM borrowers 
WHERE categorycode IN (SELECT categorycode FROM categories WHERE can_be_guarantee = 1) 
AND borrowernumber NOT IN(SELECT guarantee_id FROM borrower_relationships)
```

**Comment by IrinaHalminen** [comment]
Vaarasta löytyi näitä 3 kpl. Korjasin yhdelle oikean syntymävuosisadan. Yhden tiedot voi poistaa 12.10. (tili ollut 5 vuotta käyttämättömänä). Kolmas siirtyy aikuisasiakkaaksi marraskuussa. Kaikilla näillä oli sama Viimeksi päivitetty -ajankohta, 22.5.2023 klo 14.22.

**Comment by ReettaSiili** [comment]
Siilissä tämä raportti ei tosiaan löytänyt mitään, enkä muista että meillä olisi ollut ongelmia tässä asiassa.

**Comment by Kati-Helle** [comment]
Hellestä löytyneille lisätty asiakastietoon huoltajatiedon puuttumisesta kertova viesti Lisää viesti -toiminnolla.

**Comment by Tuomas-Kyyti** [comment]
Kyytistä löytyi 3 kpl, mutta kaikilla oli jo huomautusviesti. Näitä on siis täällä ennenkin selvitetty.

**Comment by pirliis** [comment]
OUTIssa löytyi 46 kpl. Suurimmalla osalla oli jo rajoite, jossa tieto takaajan puuttumisesta.

**Comment by ajrajala** [comment]
Vaskissa löytyi 111 kpl, mutta olivat joko kirjastojen harjoituskortteja tai sitten asiakkaille oli jo lisätty tieto takaajan puuttumisesta.

**Comment by Hanna-Lumme** [comment]
Lumpeissa löytyi 50kpl. Kaikkiin on lisätty huomautus, että takaajatieto täytyy lisätä.
